src/Compilers/Core/Portable/Operations/OperationNodes.cs

```diff
@@ -7940,26 +7953,47 @@ public override void Accept(OperationVisitor visitor)
     }
 
     /// <summary>
-    /// Represents a C# declaration pattern.
+    /// Represents a C# recursive pattern.
     /// </summary>
-    internal sealed partial class RecursivePattern : Operation, IRecursivePatternOperation
+    internal sealed partial class RecursivePatternOperation : Operation, IRecursivePatternOperation
     {
-        public RecursivePattern(ISymbol declaredSymbol, SemanticModel semanticModel, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue, bool isImplicit) :
-                    base(OperationKind.DeclarationPattern, semanticModel, syntax, type, constantValue, isImplicit)
+        public RecursivePatternOperation(ITypeSymbol inputType, ITypeSymbol matchedType, ISymbol deconstructSymbol, ImmutableArray<IPatternOperation> deconstructionSubpatterns, ImmutableArray<(ISymbol, IPatternOperation)> propertySubpatterns, ISymbol declaredSymbol, SemanticModel semanticModel, SyntaxNode syntax, bool isImplicit) :
```

Comment:
>ImmutableArray<IPatternOperation> deconstructionSubpatterns, ImmutableArray<(ISymbol, IPatternOperation)> propertySubpatterns [](start = 116, length = 125)

Nit: when I did my refactoring, I tried to put the `IOperation` children first in the arg lists.

Reply:
I'm trying to keep the order consistent whenever the parts of recursive patterns are specified separately.